pkg/systemd/logs.js

```diff
@@ -192,7 +199,59 @@ $(function() {
             }
         }
 
+        function clear_service_list() {
+            $('#journal-services-list').empty()
+                    .append($('<li>').append($('<a>')
+                            .text(_("All"))
+                            .attr('data-service', "")))
+                    .append($('<li>')
+                            .addClass("divider"));
+        }
+
+        function load_service_filters(match, options, stopped) {
+            var tags_count = 0;
+            if (stopped) {
+                options["reverse"] = true;
+                options["cursor"] = stopped;
+            }
+            procs.push(journal.journalctl(match, options)
+                    .fail(query_error)
+                    .stream(function(entries) {
+                        entries.forEach(function(entry) {
+                            current_services.add(entry["SYSLOG_IDENTIFIER"]);
+                        });
+                        tags_count += entries.length;
+                        if (tags_count >= query_count) {
+                            stopped = entries[entries.length - 1]["__CURSOR"];
+                            load_service_filters(match, options, stopped);
+                            this.stop();
+                        }
+                    })
+                    .done(show_service_filters)
+            );
+        }
+
+        function show_service_filters() {
+            clear_service_list();
+
+            // Sort and put into list
+            Array.from(current_services).sort(function (a, b) {
+                return a.toLowerCase().localeCompare(b.toLowerCase());
+            })
```

Comment:
Just mentioning, we run Babel on every file consistently now, so it's perfectly okay to use ES6 arrow functions and other modern syntax here.